flexget/plugins/sites/torrentleech.py

```diff
@@ -121,48 +120,35 @@ def search(self, task, entry, config=None):
                    quote(query.encode('utf-8')) + filter_url)
-            log.debug('Using %s as torrentleech search url' % url)
+            log.debug('Using {} as torrentleech search url'.format(url))
 
-            page = task.requests.get(url, cookies=login.cookies).content
-            soup = get_soup(page)
+            results = task.requests.get(url, cookies=login.cookies).json()
 
-            for tr in soup.find_all("tr", ["even", "odd"]):
-                # within each even or odd row, find the torrent names
-                link = tr.find("a", attrs={'href': re.compile('/torrent/\d+')})
-                log.debug('link phase: %s' % link.contents[0])
+            for torrent in results['torrentList']:
                 entry = Entry()
-                # extracts the contents of the <a>titlename/<a> tag
-                entry['title'] = link.contents[0]
-
-                # find download link
-                torrent_url = tr.find("a", attrs={'href': re.compile('/download/\d+/.*')}).get('href')
-                # parse link and split along /download/12345 and /name.torrent
-                download_url = re.search('(/download/\d+)/(.+\.torrent)', torrent_url)
-                # change link to rss and splice in rss_key
-                torrent_url = 'https://v4.torrentleech.org/rss' + download_url.group(1) + '/' \
-                              + rss_key + '/' + download_url.group(2)
-                log.debug('RSS-ified download link: %s' % torrent_url)
-                entry['url'] = torrent_url
-
-                # us tr object for seeders/leechers
-                seeders, leechers = tr.find_all('td', ["seeders", "leechers"])
-                entry['torrent_seeds'] = int(seeders.contents[0])
-                entry['torrent_leeches'] = int(leechers.contents[0])
-                entry['search_sort'] = torrent_availability(entry['torrent_seeds'], entry['torrent_leeches'])
+                entry['title'] = torrent['name']
 
-                # use tr object for size
-                size = tr.find("td", text=re.compile('([\.\d]+) ([TGMK]?)B')).contents[0]
-                size = re.search('([\.\d]+) ([TGMK]?)B', size)
-
-                entry['content_size'] = parse_filesize(size.group(0))
+                # construct download URL
+                torrent_url = 'https://www.torrentleech.org/rss/download/{}/{}/{}'.format(
+                    torrent['fid'], rss_key, torrent['filename'])
+                log.debug('RSS-ified download link: {}'.format(torrent_url))
+                entry['url'] = torrent_url
 
+                # seeders/leechers
+                entry['torrent_seeds'] = torrent['seeders']
+                entry['torrent_leeches'] = torrent['leechers']
+                entry['search_sort'] = torrent_availability(entry['torrent_seeds'],
+                                                            entry['torrent_leeches'])
+                entry['content_size'] = torrent['size']
```

Comment:
we have a util that converts size to our requested format: https://github.com/Flexget/Flexget/blob/c616922dce762fe5d26ec8b50d27b264f6e91467/flexget/utils/tools.py#L441

Reply:
The util is according to it's docstring meant to parse a string containing a size to a MiB float. The value returned from the torrentleach API is a int of bytes.

Reply:
Yes, so just append 'b' to the int. Not the nicest way and I've been wanting to change the function signature, but it's also very low priority.